front/app/components/UI/ProjectFilter/index.tsx

```diff
@@ -103,25 +109,72 @@ const ProjectFilter = ({
     localize,
   ]);
 
+  // Filter options based on search value
+  const filteredOptions = useMemo(() => {
+    if (!allProjectOptions) return [];
+    if (!searchValue) return allProjectOptions;
+
+    return allProjectOptions.filter((option) =>
+      option.label.toLowerCase().includes(searchValue.toLowerCase())
+    );
+  }, [allProjectOptions, searchValue]);
+
+  const handleInputChange = useMemo(() => {
+    return debounce((searchTerm: string) => {
+      setSearchValue(searchTerm);
+    }, 300);
+  }, []);
+
+  const handleChange = useCallback(
+    (option: Option | null) => {
+      if (!option) {
+        onProjectFilter({ value: undefined, label: '' });
+        return;
+      }
+      onProjectFilter(option);
+    },
+    [onProjectFilter]
+  );
+
   const label = formatMessage(messages.labelProjectFilter);
 
+  if (!allProjectOptions) {
+    return (
+      <Box {...boxProps}>
+        {!hideLabel && <Label>{label}</Label>}
+        <Spinner />
+      </Box>
+    );
+  }
+
+  const selectedOption =
+    allProjectOptions.find((option) => option.value === selectedProjectId) ||
+    null;
+
   return (
-    <Box {...boxProps}>
-      {projectFilterOptions ? (
-        <StyledSelect
-          id="projectFilter"
-          label={hideLabel ? undefined : label}
-          value={selectedProjectId}
-          options={projectFilterOptions}
-          placeholder={placeholder}
-          onChange={onProjectFilter}
-        />
-      ) : (
-        <>
-          {!hideLabel && <Label>{label}</Label>}
-          <Spinner />
-        </>
-      )}
+    <Box {...boxProps} width="100%">
+      <ReactSelect
```

Comment:
The `ReactSelect` solution works well, the only thing is that the UX is less consistent with the newly introduced `MultiSelect` that I introduced. There, the search bar cannot be inside of the main button the way it is here because we already need it to show the name of the filter + how much stuff is selected, so instead the search has to be inside the dropdown. So it would be more consistent if the search here was also inside of the dropdown. But then it wouldn't work out of the box like this. And I guess it works fine like this so maybe not worth worrying about